core/templates/dev/head/pages/exploration_editor/simple_editor_tab/QuestionHashService.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that contains common functionalities for creating
+unique Question identifiers.
+ */
+
+oppia.factory('QuestionHashService', [function() {
+  return {
+    getHash: function(question, subfield) {
+      var questionId;
+      var suffix;
+      if (angular.isDefined(question)) {
+        if (angular.isString(question)) {
```

Comment:
This feels weird. How can a question be either a string or an object? I mean, questions ought to be questions...

If this is to handle title/introduction, perhaps you should have different functions for these things? And then you can have a common private method that takes a string and returns s.toLowerCase().replace(' ', '-'), since that seems like common functionality.